### DISCUSSION TITLE
6차 데모데이까지의 목표에 대하여

### DISCUSSION THREAD

**Discussion by yeonkkk** [woowacourse-teams/2023-pium#396] [개발 논의]
안녕하세요 피움 여러분🌱.
지난주에 팀의 방향성을 정하고, 각자 생각을 정리한 뒤에 6차 데모데이까지 어떤 작업을 할지 오늘 함께 논의해보기로 했습니다.  
그래서 회의 전에 일정 추산 및 앞으로의 방향 결정에 참고할 만한 내용을 정리해보려고 합니다.

**1. 목표 수렴의 필요성**
남은 레벨 4 기간 동안 할 일을 크게 미션, 취준, 팀 프로젝트로 나눠볼 수 있을 것 같습니다.  
미션은 수료 조건이기 때문에 우리의 의지와 관계없이 무조건 시간을 할애해야하는 부분이고, 남은 것은 취준과 팀 프로젝트네요.  
각자 팀 프로젝트, 취준 중에 우선하고자 하는게 있을 것이라고 생각합니다.  
누구라도 취준을 우선으로 둔다면, 자연스럽게 팀 프로젝트에는 많은 시간을 할애하기 어려울 수 있다는 생각이 들어요.  
그래서  앞으로의 작업이나 일정 추산에 이 부분을 반영해야할 것 같고, 오늘 회의에서 이 부분에 대해 이야기 하면 좋을 것 같아요.    

**2. 우형 지원 기간**
앞 내용과 이어지는 이야기일 수 있는데요, 우아한형제들 지원 기간도 함께 고려해야할 것 같습니다.  
우아한형제들 지원하실 분들은 사실상 6차 데모데이 기간 동안 준비하셔야하니 이 부분도 생각해봐야할 것 같아요.  

**3. 주어진 시간**
오늘부터 다음 데모데이까지 주말과 공휴일을 제외하면 워킹데이가 약 15일 남습니다.  
6차 데모데이 요구사항 적용, 미션과 취준을 병행한다고 하면 주어진 날짜가 긴 것은 아니라서 잘 생각해봐야 할 것 같습니다.   

위 내용이 새로운 기술을 도입하거나 새로운 기능을 개발하는 것에 조금 보수적이라고 느낄 수 있으나, 팀 프로젝트를 함에 있어서 정해진 기간 내에 실현 가능한 계획을 세우고 실행하는 것 역시 중요하다고 생각되어 공유해봅니다!  
그리고 모두에게 중요한 시기인 만큼 같이 맞춰보고 가면 좋을 것 같아요.

**Comment by Choi-JJunho** [comment]
# 뭘 해보고 싶은가?

- PWA: 안해본 내용이라 일정 추산이 잘 안됨 (학습비용)
- 검색엔진: 자음 검색 하고싶음 
- 도커: 무중단 배포 과정에서 사용할 수 있음
- DB Replication: `@Transactional(readOnly = true)`
- 스프링 이벤트 & 비동기처리: 사진 업로드
  - 업로드 과정에서 이미지 최적화 과정도 고려해볼 수 있다.
- CSS 기깔나게 만들기
- 동작하는 기능에서 View 적인 요소 리팩터링
  - 페이지 이동간 애니메이션
  - nav바를 눌렀을 때 통통튀는 애니메이션

# 실제로 무엇을 할 수 있을까?

각종 일정을 고려해봤을 때 프로젝트에 할애할 수 있는 시간으로 7~ 10일정도의 일정이 나올 것으로 예상된다.


## 우선순위를 나눠봅시다.

요구사항

### 해야하는것
- 무중단 배포 및 DB Connection Pool 설정
  - 도커 도입
  - 세션 클러스터링
- 레이아웃 영향을 크게 받는 컴포넌트를 하나 정하고, 누구나 사용할 수 있게 분리하여 배포
- 이미지 최적화 서버에서 batch로 자동화

## 의견 

각자 하고싶은걸 하는데 기능에 영향이 안가는 범위에서 각자 해보는건 어떤가요?
(대신 리뷰, Merge, 지식공유 필수!)

### 백엔드

- 검색엔진: 자음 검색 하고싶음 @Choi-JJunho 
- 도커: 무중단 배포 과정에서 사용할 수 있음 @yeonkkk 
- DB Replication: `@Transactional(readOnly = true)` @rawfishthelgh 
- 스프링 이벤트 & 비동기처리: 사진 업로드 @Kim0914 
  - 업로드 과정에서 이미지 최적화 과정도 고려해볼 수 있다.

### 프론트엔드

- PWA: 안해본 내용이라 일정 추산이 잘 안됨 (학습비용) @hozzijeong 
- 유틸함수 테스트작성, 이미지요청 로직 개선, Cypress 비밀 파헤치기(?), 참Coil 만들기 @WaiNaat
- 동작하는 기능에서 View 적인 요소 리팩터링 @bassyu 
  - 웹 타겟으로 반응형 구성하기
  - 달력 컴포넌트 연도 변경이 안되는 문제 해결하기(?)
  - 페이지 이동간 애니메이션 
  - CSS 기깔나게 만들기 
  - nav바를 눌렀을 때 통통튀는 애니메이션

**Comment by Choi-JJunho** [comment]
# 프론트 사용자 피드백

- 메인 화면에서 검색 화면에 포커싱을 주었을 때 검색창에 대한 인터렉션이 있으면 좋겠다.
  - native input이 올라왔을 때 상단 화면이 변하면 좋겠다.
  - 완성도 있는 앱처럼 보이려면 최근 검색한 식물 목록 같은것도 있으면 좋겠다.

- 모두의 정원 
  - 화면에서 사진이 계속 새로 로딩됨
  - 필터가 없을 때 빈공간이 없어도 될것 같다.
  - <img width="350" alt="image" src="https://github.com/woowacourse-teams/2023-pium/assets/49794401/eb98ccd8-2e1c-444f-883a-fa98436ae2a2">
  - 모두의 정원 오른쪽하단 + 버튼 box shadow를 넣어주면 좋겠다.
  - 유저 select를 없애주면 좋을 것 같다.
  - <img width="350" alt="image" src="https://github.com/woowacourse-teams/2023-pium/assets/49794401/d2a77f91-8e52-431c-807d-94d34b840133">
  - 태그를 눌렀을 때 검색이 됐으면 좋겠다.
  - 모두의 정원 스켈레톤 폼 미쳤다 ㅇㅈㅇㅈ

- 전체적인 사진 화질이 별로다. (좀 침침하다)
  - 압축을 해도 실제로 화면에 보여줄 크기보다 width, height 두배로 설정하면 좋을 것 같다.

- 탭 이동이 너무 딱딱해보인다.
  - 하단바에는 슬라이드 액션이 있으면 좋겠다. 

- 새로고침하다가 리마인더를 처음 들어간 순간 이미지가 엄청 깨진다.
  - 이미지 로드가 너무 느리다..

- 내 식물 갔다가 마이페이지로 갔을 때 너무 짧은 로딩 스피너가 UX가 안좋아보인다.
  - 하단바도 같이 안보인다.
  - 하단바를 공통으로 쓰게 라우팅 바깥으로 꺼내주면 좋겠다.
    - 하단바가 모든 페이지에 하나씩 들어가있는 느낌이다.
  - ![KakaoTalk_Video_2023-09-26-00-18-21](https://github.com/woowacourse-teams/2023-pium/assets/49794401/9db10acb-d2fe-442b-8d25-851523ac0d2b)

- (선택) 하단 바 바로 위에 반투명한 영역을 두면 좋을 것 같다.
  - <img width="350" alt="image" src="https://github.com/woowacourse-teams/2023-pium/assets/49794401/fc46d12e-5673-41c6-a3f1-b3b5ba3ad37e">

**Reply by WaiNaat** [reply]
> 하단바가 모든 페이지에 하나씩 들어가있는 느낌이다.

아니 이거 비밀 어떻게푸신거죠??

**Reply by Choi-JJunho** [reply]
그저 추측이였다고합니다

**Reply by WaiNaat** [reply]
### 프론트 회의 결과

#### 검색창 클릭 시 "검색 모드"  + 지난 검색어

- 새로운 페이지 개발이라 시간 소요 예상
- 검색 자체가 피움의 메인 기능은 아님
- 일단 보류
 
#### 화면에서 사진이 계속 새로 로딩되는 현상

- 인지 중인 문제 (참새 담당으로 되어있음)\
- 주노가 30분에 한번씩 최적화 진행해서 많이 개선
- 당선

#### 모두의 정원 필터 없을때 빈공간 없애기

- 당선

#### 모두의 정원 우하단 버튼 box shadow

- 마이페이지 문의하기도 같이 처리하기
- 당선

#### 유저 select를 없애주세요

- 왜 막아야 하죠?
    - 중복 닉네임 방지?
    - 좀 더 앱 같은 느낌을 위해?
    - 애초에 드래그할 이유가 없어서?
- 일단 보류

#### 모두의 정원 태그 눌렀을 때 검색

- 난이도 추천은 새로운 기능이라 패스
- 식물 종류는 클릭했을 때 검색
- 당선 (후순위)

#### 전체적인 사진 화질이 좀 별로다

해결 (고마워요 주노)

#### 하단바 슬라이드 액션

- 당선

#### 이미지 로드가 너무 느리다

해결 (고마워요 주노)

#### 너무 짧은 로딩 스피너

- 당선

#### 하단 바 바로 위에 반투명한 영역

- 사전 식물 상세 정보에서만 적용 (NavBar는 아님)
- 당선

**Reply by hozzijeong** [reply]
+ 토스트 UI에 그자 초과됐을 경우 어떻게 처리할 것인지?

**Comment by Choi-JJunho** [comment]
6차 데모데이 완료로 close하겠습니다
고생하셨습니다.
